File: src/main/org/nlogo/prim/etc/_show.java
package org.nlogo.prim.etc;

import org.nlogo.api.LogoException;
import org.nlogo.nvm.Command;
import org.nlogo.api.Syntax;
import org.nlogo.nvm.Workspace;

public final strictfp class _show
    extends Command {
  @Override
  public void perform(final org.nlogo.nvm.Context context) throws LogoException {
    workspace.outputObject
        (args[0].report(context),
            context.agent,
            true, true,
            Workspace.OutputDestination.NORMAL);
    context.ip = next;
  }

  @Override
  public Syntax syntax() {
    int[] right = {Syntax.WildcardType()};
    return Syntax.commandSyntax(right);
  }

  ///////////////////////////////////////////////
  // for use with bytecode generator

  public void perform_1(final org.nlogo.nvm.Context context, Object arg0) throws LogoException {
    workspace.outputObject(arg0, context.agent, true, true,
        Workspace.OutputDestination.NORMAL);
    context.ip = next;
  }

}

File: netlogo-gui/src/main/prim/etc/_show.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.api.OutputDestination
import org.nlogo.nvm.{ Command, Context }

class _show extends Command {

  override def perform(context: Context): Unit = {
    workspace.outputObject(
      args(0).report(context), context.agent,
      true, true, OutputDestination.Normal)
    context.ip = next
  }
}
